exonum/src/sandbox/consensus/round_details.rs

```diff
@@ -685,6 +685,156 @@ fn handle_precommit_incorrect_txs() {
     // Here majority of precommits is achieved and node should panic.
 }
 
+/// Scenario for this test is similar to the `handle_precommit_incorrect_txs`
+/// but node receives incorrect tx before propose.
+///
+/// Here, node receives majority of precommits for a block with incorrect tx.
+///
+/// Normally, after receiving all the transactions for a propose, node should send a prevote for it.
+/// In our case, propose contains the incorrect tx, so we expect node **NOT** to vote for it.
+/// Later, when majority of nodes will send precommits (meaning that they agree with propose),
+/// node should panic because it doesn't agree with the block being accepted.
+#[test]
+#[should_panic(expected = "handle_majority_precommits: propose contains")]
+fn handle_precommit_incorrect_tx_received_before_propose() {
```

Comment:
As with the test in the `block_request` module, it could make sense to unite the logic in this test with `handle_precommit_incorrect_txs`.